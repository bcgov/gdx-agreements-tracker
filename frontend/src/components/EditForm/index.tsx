import React from "react";
import { Box, Button } from "@mui/material";
import { Form, Formik } from "formik";
import { FormLayout } from "../GDXForm/FormLayout";
import { FormInput } from "../FormInput";
import { IEditFields } from "../../types";

export const EditForm = ({
  initialValues,
  onSubmit,
  editFields,
}: {
  initialValues: any;
  onSubmit: any;
  editFields: IEditFields[];
}) => {
  return (
    <Formik initialValues={initialValues} onSubmit={onSubmit}>
      {({ setFieldValue, values, handleChange, dirty }) => {
        return (
          <Form>
            <FormLayout>
              {editFields.map(({ fieldName, fieldType, fieldLabel, width, tableName }) => {
                return (
                  <FormInput
                    setFieldValue={setFieldValue}
                    fieldValue={values?.[fieldName]}
                    fieldName={fieldName}
                    fieldType={fieldType}
                    fieldLabel={fieldLabel}
                    handleChange={handleChange}
                    width={width}
                    key={fieldName}
                    tableName={tableName}
                  />
                );
              })}
            </FormLayout>
            <Box m={1} display="flex" justifyContent="flex-end" alignItems="flex-end">
              <Button
                type="submit"
                variant="contained"
                color="success"
                disabled={dirty ? false : true}
              >
                Submit
              </Button>
            </Box>
          </Form>
        );
      }}
    </Formik>
  );
};
