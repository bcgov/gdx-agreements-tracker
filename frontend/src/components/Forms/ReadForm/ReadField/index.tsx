import {
  Checkbox,
  Chip,
  Divider,
  FilledInput,
  FormControl,
  InputLabel,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  ListSubheader,
  Select,
  Stack,
  TextField,
  Typography,
} from "@mui/material";
import { IReturnValue } from "types";
import { GridItem } from "../../FormLayout/GridItem";
import { useNavigate } from "react-router-dom";
import DescriptionIcon from "@mui/icons-material/Description";
/**
 * Represents a read-only field component.
 *
 * @param   {object}       props        - The properties object.
 * @param   {string}       props.width  - The width of the field.
 * @param   {string}       props.title  - The title of the field.
 * @param   {IReturnValue} props.value  - The value to display.
 * @param   {string}       [props.type] - The type of the field.
 * @returns {JSX.Element}               The rendered ReadField component.
 */

export const ReadField = ({
  width,
  title,
  value,
  type,
}: {
  width: string;
  title: string;
  value: IReturnValue;
  type?: string;
}) => {
  const navigate = useNavigate();
  const renderer = () => {
    switch (type) {
      case "multiSelect":
        return (
          <FormControl fullWidth>
            <InputLabel>{title}</InputLabel>
            <Select
              disabled
              multiple
              value={value}
              IconComponent={() => null} //To hide dropdown arrow
              input={
                <FilledInput
                  sx={{
                    "& .MuiInputBase-input.Mui-disabled": {
                      WebkitTextFillColor: "#333",
                      background: "#fff",
                    },
                    "& .MuiInputBase-root": {
                      WebkitTextFillColor: "#333",
                      background: "#fff!important",
                    },
                  }}
                />
              }
              renderValue={(selected) => (
                <Stack gap={1} direction="row" flexWrap="wrap">
                  {(value as string[]).map((value) => (
                    <Chip key={value} label={value} />
                  ))}
                </Stack>
              )}
            ></Select>
          </FormControl>
        );
      case "checkbox":
        return (
          <div>
            <Typography variant="subtitle1" color="textSecondary">
              {title}:
            </Typography>
            <Checkbox disabled checked={value as boolean} />
          </div>
        );

      case "link":
        return (
          <List
            sx={{ width: "100%", maxWidth: "13rem", bgcolor: "#fbfbfb" }}
            subheader={
              <ListSubheader sx={{ bgcolor: "#2d2b2b", color: "#fff" }}>{title}</ListSubheader>
            }
          >
            {(value as unknown as { link: string; label: string }[]).map((value) => {
              return (
                <>
                  <Divider />
                  <ListItem component="div" disablePadding>
                    <ListItemButton onClick={() => navigate(value.link)}>
                      <ListItemText sx={{ color: "blue" }} primary={`${value.label}`} />
                      <ListItemIcon sx={{ color: "inherit" }}>
                        <DescriptionIcon />
                      </ListItemIcon>
                    </ListItemButton>
                  </ListItem>
                </>
              );
            })}
          </List>
        );

      default:
        return (
          <TextField
            disabled
            label={title}
            value={value}
            fullWidth
            sx={{
              "& .MuiInputLabel-root ": {
                WebkitTextFillColor: "#16008fc4",
              },
              "& .MuiInputBase-input.Mui-disabled": {
                WebkitTextFillColor: "#333",
                background: "#fff",
              },
              "& .MuiInputBase-root": {
                WebkitTextFillColor: "#333",
                background: "#fff!important",
              },
            }}
            id="filled-size-small"
            variant="filled"
            size="small"
            multiline
          />
        );
    }
  };

  return <GridItem width={width}>{renderer()}</GridItem>;
};
