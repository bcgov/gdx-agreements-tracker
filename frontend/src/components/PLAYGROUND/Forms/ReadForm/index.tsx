import React from "react";

export const ReadForm = () => {
  return <div>ReadForm</div>;
};
