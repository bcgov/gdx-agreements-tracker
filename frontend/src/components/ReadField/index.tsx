import styled from "@emotion/styled";
import { Paper } from "@mui/material";
import React from "react";
import { GridItem } from "../GDXForm/FormLayout/GridItem";

export const ReadField = ({
  width,
  title,
  value,
}: {
  width: string;
  title: string;
  value: string | number;
}) => {
  const StyledPaper = styled(Paper)({
    height: "auto",
    flex: 1,
    display: "flex",
    overflow: "auto",
  });

  return (
    <GridItem width={width}>
      <StyledPaper>
        <h3>{title}</h3>
        <p>{value}</p>
      </StyledPaper>
    </GridItem>
  );
};
