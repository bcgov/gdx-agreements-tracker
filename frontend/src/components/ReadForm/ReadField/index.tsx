import styled from "@emotion/styled";
import { Checkbox, Paper } from "@mui/material";
import React from "react";
import { IReturnValue } from "types";
import { GridItem } from "../../GDXForm/FormLayout/GridItem";

export const ReadField = ({
  width,
  title,
  value,
}: {
  width: string;
  title: string;
  value: IReturnValue;
}) => {
  const renderer = () => {
    switch (typeof value) {
      case "boolean":
        return (
          <>
            <h3>{title}</h3>
            <Checkbox disabled checked={value} />
          </>
        );
      default:
        return (
          <>
            <h3>{title}</h3>
            <p>{Array.isArray(value) ? value.join(" | ") : value}</p>
          </>
        );
    }
  };

  const StyledPaper = styled(Paper)({
    height: "auto",
    flex: 1,
    display: "flex",
    overflow: "auto",
  });

  return (
    <GridItem width={width}>
      <StyledPaper>{renderer()}</StyledPaper>
    </GridItem>
  );
};
