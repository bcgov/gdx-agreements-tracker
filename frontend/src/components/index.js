export { ButtonNav } from "./ButtonNav";
export { Debug } from "./Debug";
export { Loader } from "./Loader";
export { Main } from "./Layout/Main";
export { PageLink } from "./PageLink";
export { RowEditDeleteChip } from "./RowEditDeleteChip";
export { Sidebar } from "./Sidebar";
export { DetailsList } from "./DetailsList";
export { Table } from "./Table";
