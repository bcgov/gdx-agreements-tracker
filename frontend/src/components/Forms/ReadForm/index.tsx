import React from "react";
import { IReturnValue } from "types";
import { ReadField } from "./ReadField";
import { FormLayout } from "components/Forms/FormLayout";

export const ReadForm = ({
  fields,
}: {
  fields: { width: string; title: string; value: IReturnValue }[];
}) => {
  return (
    <FormLayout>
      {fields.map(({ width, title, value }) => {
        console.log('value', value)
        return <ReadField width={width} title={title} value={value} key={title}></ReadField>;
      })}
    </FormLayout>
  );
};
