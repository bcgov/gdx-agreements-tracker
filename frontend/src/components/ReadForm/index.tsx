import React from "react";
import { FormLayout } from "../GDXForm";
import { ReadField } from "./ReadField";

export const ReadForm = ({
  fields,
}: {
  fields: { width: string; title: string; value: string | number }[];
}) => {
  return (
    <FormLayout>
      {fields.map(({ width, title, value }) => {
        return <ReadField width={width} title={title} value={value}></ReadField>;
      })}
    </FormLayout>
  );
};
