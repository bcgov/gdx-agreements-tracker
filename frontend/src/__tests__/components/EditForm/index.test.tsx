import { QueryClient, QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { EditForm } from "../../../components/EditForm";
import TestRenderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import { mount, shallow } from "enzyme";
import { ReadField } from "components/ReadForm/ReadField";
import { IEditFields } from "types";

const editFields: () => IEditFields[] = () => {
  return [
    {
      fieldName: "mock_field",
      fieldLabel: "Mock Field",
      fieldType: "singleText",
      width: "full",
    },
  ];
};

const initialValues = { mock_field: "mock value" };

const mockFunction = jest.fn();

// jest.mock("@react-keycloak/web", () => ({
//     FormInput: () => (<Fi>{}</Fi>),
// }));

describe("Renders <EditForm /> component", () => {
  const wrapper = shallow(
    <EditForm initialValues={initialValues} onSubmit={mockFunction} editFields={editFields()} />
  );

  it("renders the EditForm component", () => {
    expect(wrapper.html()).toMatchSnapshot();
    // expect((wrapper.props().children[0].props.title)).toEqual("Mock Title");
  });
});
