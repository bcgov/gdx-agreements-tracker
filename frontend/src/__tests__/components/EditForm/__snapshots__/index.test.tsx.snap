// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <EditForm /> component renders the EditForm component 1`] = `"<form action=\\"#\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-1idn90j-MuiGrid-root\\"><div class=\\"MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root\\" role=\\"mock_field_input\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"true\\" for=\\"mock_field\\" id=\\"mock_field-label\\">Mock Field</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"mock_field\\" name=\\"mock_field\\" value=\\"mock value\\" class=\\"MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-14lo706\\"><span>Mock Field</span></legend></fieldset></div></div></div></div><div class=\\"MuiBox-root css-y72vsk\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\">Submit</button></div></form>"`;
